### PR TITLE
allow embeddings vector to be used for mmr searching (#2620)

### LINKED ISSUE #2620: Feature: Allow embedding vector search for max_marginal_relevance_search
**Describe the feature**: Currently there is no option to perform the max_marginal_relevance_search with only query embeddings vector. The query is mandatory and because of which the embedding service needs to be passed.

### PR DESCRIPTION
**Pull request by rishabh208gupta**
Closes #2620 
Provide an option to pass vector embeddings to search for in the [max_marginal_relevance_search](https://github.com/elastic/elasticsearch-py/blob/6521b55761ccb6e7033ea643037fd1c94ea818c4/elasticsearch/helpers/vectorstore/_async/vectorstore.py#L344-L357)
This gives us the flexibility to perform mmr search with either a text query or an embeddings vector

### CODE REVIEW

**Comment by rishabh208gupta** [comment]
I have signed the document, I don't get it why its complaining 
[Uploading Please_DocuSign_Individual_Contributor_Licen.pdf…]()

File: elasticsearch/helpers/vectorstore/_async/vectorstore.py [hunk 1]
```diff
@@ -344,8 +344,8 @@ async def _create_index_if_not_exists(self) -> None:
     async def max_marginal_relevance_search(
         self,
         *,
-        embedding_service: AsyncEmbeddingService,
```
Comment (pquentin) [inline elasticsearch/helpers/vectorstore/_async/vectorstore.py:347]: Oh, I did not realize this was not useful. Unfortunately, you can't remove it as the LangChain integration passes this parameter: https://github.com/langchain-ai/langchain-elastic/blob/9915f1e16d20e992896a651f7a3ec94db56a76f0/libs/elasticsearch/langchain_elasticsearch/vectorstores.py#L816-L825

**Review by pquentin — COMMENTED** [review]
Thanks! Can you please add to the test a call with query_embedding passed as a parameter?

File: elasticsearch/helpers/vectorstore/_async/vectorstore.py (hunk 1 above)
Comment (rishabh208gupta) [reply elasticsearch/helpers/vectorstore/_async/vectorstore.py:347]: Okay I will add it as an optional param, my plan was to create another PR in the langchain repo.

**Comment by pquentin** [comment]
Hey, the CLA is complaining because your patch is authored from "Rishabh Gupta <rishabhgupta@Rishabhs-MacBook-Pro.local>". You need to setup an user email and name in git, then `git commit --amend --reset-author` and force push.

**Comment by rishabh208gupta** [comment]
Thanks @pquentin !

**Comment by pquentin** [comment]
buildkite test this please

**Review by miguelgrinberg — APPROVED** [review]
LGTM

**Comment by pquentin** [comment]
buildkite test this please

**Pull request merged**